src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -245,6 +252,22 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
       });
   };
 
+  const setDefaultDataSource = async () => {
+    try {
+      for (const dataSource of selectedDataSources) {
+        if (uiSettings.get('defaultDataSource') === dataSource.id) {
+          await handleSetDefaultDatasourceAfterDeletion(savedObjects.client, uiSettings);
+        }
+      }
+    } catch (e) {
+      setIsLoading(false);
+      handleDisplayToastMessage({
+        id: 'dataSourcesManagement.editDataSource.setDefaultDataSourceFailMsg',
+        defaultMessage: 'Unable to set the Data Source to default. Please try it again.',
```

Comment:
How does index pattern handle the error? How would user be able to retry if the data source is already deleted?

Reply:
Let me chanke the default message to "Unable to find a default datasource. Please set a new default datasource."